cmd/changelog-parser/main.go

```diff
@@ -272,6 +287,19 @@ func runParser(options cliOptions) {
 	log.Printf("Changelog parser completed!")
 }
 
+// If no filename is provided, we generate one based on the file type
+func (options *cliOptions) generateOutputFilename() {
+	if options.OutputFilename == "" {
+		if options.OutputType == "unreleased" {
+			options.OutputFilename = templates[options.OutputType].OutputName
+		} else {
+			options.OutputFilename = fmt.Sprintf(
+				templates[options.OutputType].OutputName,
+				options.Version)
+		}
```

Comment:
Once you start adding logic to a struct, it's time to move it out into its own package/module usually, especially if that struct is in `main.go`. Once you move this out, this should be changed to `OutputFilename()`.